jolt-core/src/jolt/vm/bytecode.rs

```diff
@@ -174,15 +173,14 @@ impl<F: PrimeField, G: CurveGroup<ScalarField = F>> BytecodePolynomials<F, G> {
     pub fn new(mut bytecode: Vec<ELFRow>, mut trace: Vec<ELFRow>) -> Self {
         Self::validate_bytecode(&bytecode, &trace);
         Self::preprocess(&mut bytecode, &mut trace);
+        let max_bytecode_address = bytecode.iter().map(|instr| instr.address).max().unwrap();
 
         // Preprocessing should deal with padding.
         assert!(is_power_of_two(bytecode.len()));
         assert!(is_power_of_two(trace.len()));
 
         let num_ops = trace.len().next_power_of_two();
-        let code_size = bytecode.len().next_power_of_two();
-
-        println!("BytecodePolynomials::new num_ops {num_ops} code_size {code_size}");
+        let code_size = (max_bytecode_address + 1).next_power_of_two();
```

Comment:
Let's stick with pre-processing (`Self::preprocess` above) doing all padding and not adjust trace or num_ops sizes here implicitly. (I made this issue in an earlier PR)

Reply:
this isn't pre-processing; the `+ 1` here is because the bytecode addresses are 0-indexed

Reply:
I'll add a comment